### PR TITLE
Fix remove Group Member in Identity Group not working

### LINKED ISSUE #5464: Cannot remove a Member Group from an Identity Group
**Describe the bug**
Cannot remove a Member Group from an Identity Group

**To Reproduce**
Using the Vault UI, but I see the same behaviour in API.
* Create an Identity Group with Name 'Child Group'
* Create an Identity Group with Name 'Parent Group'
* Add 'Child Group' ID to 'Parent Group' Member Group List
* Verify 'Parent Group' has 'Child Group' as Member -> OK
* Remove 'Child Group' from Member Group of 'Parent Group? -> NOK

**Expected behavior**
After Removing the 'Child Group' from 'Parent Group' it should no longer be in the 'Parent Group'

**Environment:**
* Vault Server Version (retrieve with `vault status`):  0.11.2
* Vault CLI Version (retrieve with `vault version`): 0.11.2
* Server Operating System/Architecture: Vault Docker Image in Kubernetes Cluster

**Additional context**
Adding / Removing of Entities to a Identity Group works as expected
I think in https://github.com/hashicorp/vault/blob/master/vault/identity_store_util.go#L863 you never put back the 'memberGroupIDs' into the 'group' Object

### PR DESCRIPTION
**Pull request by splattner**
My proposal to fix #5464

### CODE REVIEW

File: vault/identity_store_util.go [hunk 1]
```diff
@@ -931,6 +931,59 @@ func (i *IdentityStore) sanitizeAndUpsertGroup(ctx context.Context, group *ident
...
+		removed := true
+		for _, memberGroupID := range memberGroupIDs {
+			if memberGroupID == currentMemberGroupID {
+				removed = false
+				break
+			}
+		}
+
+		// Remove ParentGroup Entry for this group from removed Group
+		if removed {
```
Comment (vishalnayak) [inline vault/identity_store_util.go]: You can remove the above `for` loop and use `!strutil.StrListContains(memberGroupIDs, currentMemberGroupID)` in place of `removed`.

File: vault/identity_store_util.go [hunk 2]
```diff
@@ -931,6 +931,59 @@ func (i *IdentityStore) sanitizeAndUpsertGroup(ctx context.Context, group *ident
...
+			for i, v := range currentMemberGroup.ParentGroupIDs {
```
Comment (vishalnayak) [inline vault/identity_store_util.go]: This `for` loop can be substituted with `currentMemberGroup.ParentGroupIDs = strutil.StrListDelete(currentMemberGroup.ParentGroupIDs, group.ID)`.

File: vault/identity_store_util.go (hunk 1 above)
Comment (splattner) [reply vault/identity_store_util.go]: thx changed this

File: vault/identity_store_util.go (hunk 2 above)
Comment (splattner) [reply vault/identity_store_util.go]: thx, I changed this

**Comment by splattner** [comment]
Thx for your Hint with the strutil, thats way nicer ;)

**Comment by vishalnayak** [comment]
I will be writing a test for this in a separate PR.

**Pull request merged**